clients/drcachesim/tools/filter/encodings2regdeps_filter.h

```diff
@@ -67,27 +72,78 @@ class encodings2regdeps_filter_t : public record_filter_t::record_filter_func_t
     parallel_shard_init(memtrace_stream_t *shard_stream,
                         bool partial_trace_filter) override
     {
-        return nullptr;
+        per_shard_t *per_shard = new per_shard_t;
+        per_shard->output_syscall_func_markers = false;
+        return per_shard;
     }
 
     bool
     parallel_shard_filter(
         trace_entry_t &entry, void *shard_data,
         record_filter_t::record_filter_info_t &record_filter_info) override
     {
+        /* Get per_shard private data.
+         */
+        per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+
+        /* Get data from record_filter.
+         */
         std::vector<trace_entry_t> *last_encoding = record_filter_info.last_encoding;
         void *dcontext = record_filter_info.dcontext;
 
-        /* Modify file_type to regdeps ISA, removing the real ISA of the input trace.
-         */
         trace_type_t entry_type = static_cast<trace_type_t>(entry.type);
         if (entry_type == TRACE_TYPE_MARKER) {
             trace_marker_type_t marker_type =
                 static_cast<trace_marker_type_t>(entry.size);
-            if (marker_type == TRACE_MARKER_TYPE_FILETYPE) {
+            switch (marker_type) {
+            /* Modify file_type to regdeps ISA. Removes the real ISA of the input
+             * trace and adds OFFLINE_FILE_TYPE_ARCH_REGDEPS.
+             */
+            case TRACE_MARKER_TYPE_FILETYPE: {
                 uint64_t marker_value = static_cast<uint64_t>(entry.addr);
                 marker_value = update_filetype(marker_value);
                 entry.addr = static_cast<addr_t>(marker_value);
+                return true;
+            }
+            /* Output TRACE_MARKER_TYPE_FUNC_[ID | ARG | RETVAL] for SYS_futex syscalls.
```

Comment:
This doesn't seem to belong in a filter that transforms the encodings.  I would expect it to be in the type filter or possibly a new filter.

Reply:
You're right, the thing is that this is such a specific type of filtering where the only use-case (at least for now) is the release of public traces.
The type_filter would need major changes to do this type of filtering, as we're not simply listing the TRACE_MARKER_TYPE_ to drop, we're dropping TRACE_MARKER_TYPE_FUNC_ID depending on their value, and on top of that we have "state" to drop other, related markers, such as TRACE_MARKER_TYPE_FUNC_ARG, depending on the value of TRACE_MARKER_TYPE_FUNC_ID. It's probably best not to overcomplicate type_filter.
I think having a new "function_marker_filter" that focuses on TRACE_MARKER_TYPE_FUNC_ only and generalizes this a bit (listing the #function to keep, rather than hardcoding SYS_futex) is the way to go.

Reply:
SGTM.  If we had #6821 the user could get the syscall nums that were traced; could cmp to the syscall lists for the supported arches and figure out which is futex -- with corner cases of colliding numbers...maybe need an offline discussion on how to solve that.

Reply:
If we leave the "burden" of specifying directly the syscall number for the TRACE_MARKER_TYPE_FUNC_ID value to the user we don't even need #6821 for now (it can be inferred by OFFLINE_FILE_TYPE_ARCH_ of the trace right?).
If we want to be more user-friendly (having something like `--function_marker_filter_func_ids_to_keep=SYS_futex,...` rather than `--function_marker_filter_func_ids_to_keep=202,...` for an OFFLINE_FILE_TYPE_ARCH_X86_64 trace), then we probably need to "synthesize a string name for the syscall" as described in #6821.